### PR TITLE
Upgrade Dependencies, Improve CI, Fix Storybook addons

### PR DESCRIPTION
**Pull request by blaketarter**
## Description

- Upgraded all dependencies (that I could) to the latest versions. I avoided a few upgrades that broke watch mode in Expo.
- Added more tests to CI to ensure expo is setup and working correctly
- (Attempted to) Fix addons in Storybook

Attempted to fix (but didn't) #13 
Related to #6 

## 🔎 Reviewer Checklist

> Checked off by the PR **Reviewers**

### Required

> These always need to be checked

- [x] Merge destination is correct
- [x] Code is correct as understood and conforms to quality standards
- [x] Tests have been added where appropriate (unit, visual, end-to-end)
- [x] Acceptance Criteria have been met

---

>### Roles & Responsibilities
>
>#### 👨‍💻 Assignee
>
>- Initiator of this PR (be sure to set in GitHub UI)
>- Addresses feedback and change requests
>- Merges PR once approved (usually deletes branch unless `develop` or `release`)
>
>#### 👩‍💻 Reviewer
>
>- Invited to review PR by **Assignee** (via GitHub UI)
>- Is expected to complete a review and address followup

### CODE REVIEW

File: storybook/index.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
-import { getStorybookUI, configure } from "@storybook/react-native"
+import { configure, getStorybookUI } from "@storybook/react-native"
 import "loki/configure-react-native"
+import "./rn-addons"
 
 configure(() => {
   import("../src/stories")
```
Comment (roblafeve) [inline storybook/index.ts:6]: Probably old-news, but curious why stories have to be manually aggregated 🤔

File: src/components/App/index.tsx [hunk 2]
```diff
@@ -1,15 +1,15 @@
 import React from "react"
-import { StyleSheet, Text, View } from "react-native"
+import { SafeAreaView, StyleSheet, Text } from "react-native"
 import { Counter } from "../Counter"
 
 export function App() {
   return (
-    <View style={styles.container}>
+    <SafeAreaView style={styles.container}>
```
Comment (roblafeve) [inline src/components/App/index.tsx:7]: Very cool 👌

**Review by roblafeve — APPROVED** [review]
Great work 👍

File: storybook/index.ts (hunk 1 above)
Comment (blaketarter) [reply storybook/index.ts:6]: `require.context` doesn't work in a React Native environment so you have to use `import` 😞

Comment (roblafeve) [reply storybook/index.ts:6]: This sounds vaguely familiar 🙆‍♂

**Pull request merged**